app/views/shared/_certificate_details.html.erb

```diff
@@ -2,7 +2,7 @@
   <dt class="govuk-visually-hidden govuk-body-s">Environment</dt>
   <dd><%= certificate.component.environment.capitalize %></dd>
   <dt class="govuk-visually-hidden">Component</dt>
-  <dd><%= certificate.component.display %></dd>
+  <dd><%= certificate.component.display == COMPONENT_TYPE::SP_LONG ? t("user_journey.component_name.SP").capitalize : certificate.component.display %></dd> 
```

Comment:
shouldn't it be checking against the `type` instead of the display value?
```suggestion
  <dd><%= certificate.component.type == COMPONENT_TYPE::SP_LONG ? t("user_journey.component_name.SP").capitalize : certificate.component.display %></dd> 
```

Reply:
yes it shouldddd